### DISCUSSION TITLE
How to suspend a user from login?

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#2168] [Q&A] [answered]
I need to retain user informaiton and need to suspend user from login. So later require particular user can be changed to active status. How to achieve this?

**Comment by ganigeorgiev** [comment, marked as answer]
There is no global toggle for this out of the box.

If you use PocketBase as framework, you can disable the email login for a particular user using the `OnRecordBeforeAuthWithPasswordRequest` and return an error (or define a global middleware to intercept the `/api/collection/users/auth-with-password` call).

```go
app.OnRecordBeforeAuthWithPasswordRequest().Add(func(e *core.RecordAuthWithPasswordEvent) error {
	if e.Record != nil && e.Record.Email() == "block@example.com" {
		return apis.NewForbiddenError("You are not allowed to perform this action", nil)
	}

	return nil
})
```

_If you want to disable also the OAuth2 logins for accounts with the specified email, you can make use of the more generic `OnRecordAuthRequest` hook (it executes after the authentication and right before returning the token response)._

**Reply by iamdineshkumar** [reply]
I am thinking about having a bool field status in user collection which hold true(Active user) or false(Suspend). In case true then can be allowed to login else need to disallow. How to achive this? Whether e.Record will have all details?

**Reply by ganigeorgiev** [reply]
> Whether e.Record will have all details?

Yes, you can access the record fields using the [Get*/Set helpers](https://pocketbase.io/docs/record-methods/#getset-record-fields), eg.:

```go
app.OnRecordBeforeAuthWithPasswordRequest().Add(func(e *core.RecordAuthWithPasswordEvent) error {
    if e.Record != nil && !e.Record.GetBool("yourBoolField") {
        return apis.NewForbiddenError("You are not allowed to perform this action", nil)
    }

    return nil
})
```
